src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PackageBuilder.cs

```diff
@@ -893,7 +910,24 @@ private ZipArchiveEntry CreateEntry(ZipArchive package, string entryName, Compre
         private ZipArchiveEntry CreatePackageFileEntry(ZipArchive package, string entryName, DateTimeOffset timeOffset, CompressionLevel compressionLevel)
         {
             var entry = package.CreateEntry(entryName, compressionLevel);
-            entry.LastWriteTime = timeOffset;
+
+            // Here added 3 days, just in case to avoid any possible TimeZone issue, otherwise we hit this when repackaging the files.
+            if (timeOffset < ZipFormatMinDate.AddDays(3))
```

Comment:
As mentioned in [this comment](https://github.com/NuGet/NuGet.Client/pull/3793#discussion_r542736244), I disagree with this. Translate the timestamp into UTC (or use [the UTC version of getting the timestamp](https://docs.microsoft.com/en-us/dotnet/api/system.io.filesysteminfo.lastwritetimeutc?view=net-5.0)), and then timezones don't matter.  We know what the ZIP file format supports, so we can clip to the exact value.

Also, as Jon mentioned, the ZIP format also has a max date that it supports. You should add an additional case for that as well.

Reply:
Ok. Changed to UTC version and also added max date support.